### ISSUE TITLE
Content binding selects all descendants, but should only select first children.

### ISSUE THREAD

**Issue by treshugart** [skatejs/shade#1]

**Comment by antitoxic** [comment]
`>` css selector is not supported in `querySelector` yet so you can't do `customElement.querySelectorAll(' > '+contentSelectAttrValue)`. You can however do the following:
1. Create a `<div class="super-specific-random-generated-class">` and append all contents of the custom element to it. 
2. Do: `customElement.querySelectorAll('.super-specific-random-generated-class > '+contentSelectAttrValue)`

**Comment by treshugart** [comment]
That's an interesting idea. I'd be interested to see performance benchmarks between that and calling `matches()` on each child node. From my experience the latter is very performant even in IE.

**Comment by antitoxic** [comment]
I didn't want to deal with prefixes, that's why i chose this. I also improved the scenario above:
1. Don't create a div. Simple set attribute to the custom element (like `shadeid="uinquieid"`)
2. Query the parent, not the element via `parentNode.querySelectorAll('[shadid="uniqueiid"] > ' + selector)`

There's no wrapping this way.

I think simply doing this is a very clean solution. I see that others doing similar stuff but instead an attribute, they replace the ID and restore the initial value after querying - this however looks like more work and might introduce some unstyled flash of content.

**Comment by antitoxic** [comment]
Performance-wise I cant give any valuable input. I am guessing that `matches()` is a better path.

**Comment by treshugart** [comment]
I think you're right, using a unique attribute would be better. If they've attached styles or behaviour you could get FOUC, jank and / or race conditions.

**Comment by antitoxic** [comment]
I gave up on the `shadeid` solution. Nodes don't have parent when created via `document.createElement`.

I'm now using:

``` js
var navtiveMatchesSelector = el.matches || el.matchesSelector || el.webkitMatchesSelector || el.mozMatchesSelector || el.msMatchesSelector || el.oMatchesSelector;

function matchesSelector(el, selector) {
    return navtiveMatchesSelector.call(el, selector);
}

function findMatchingChildren(el, selector) {
    var matching = [];
    for (var i = 0; i < el.childNodes.length; i++) {
        var node = el.childNodes[i];
        if (node.nodeType != Node.ELEMENT_NODE)  continue;
        if (matchesSelector(node, selector)) matching.push(node);
    }
    return matching;
}
```

I see skate accounts for some kind of bug in IE9? - https://github.com/skatejs/skatejs/blob/8068f36b5f8439c54a78d54c450f53cd68043893/src/lifecycle.js#L26

**Comment by treshugart** [comment]
Yeah, that one was pretty fun. Are you using this in Shade? If so, mind patching :)

**Comment by antitoxic** [comment]
I'm not using it yet. I'm exhibiting weird bugs that don't happen with my improvised code. I can't investigate right now but will feedback later.

**Comment by antitoxic** [comment]
@treshugart to be honest I'm not very well navigating through  the code. I don't know what does what. Also I haven't been digging into Babel and new ECMA besides some reading.

It's hard for me to do pull requests even if I wanted to.

**Comment by treshugart** [comment]
Ok, let me know if you want to but just need a hand grokking the code / build.
